laplace/curvature/__init__.py

```diff
@@ -13,5 +13,5 @@
     logging.info('asdfghjkl backend not available.')
 
 __all__ = ['CurvatureInterface', 'GGNInterface', 'EFInterface',
-           'BackPackInterface', 'BackPackGGN', 'BackPackEF',
+           'BackPackInterface', 'BackPackGGN', 'BackPackEF', 'BackPackGP',
```

Comment:
Why do the GP quantities require a new backend and cannot be added to `BackPackInterface`?

Reply:
I opted for having a separate class `BackPackGP`, since if we were to move gp methods (`gp`, `k_b_b` etc.) to `BackPackInterface` class, they would be available also in `BackPackGGN` and `BackpackEF` classes even though GP has nothing to do with neither GGN nor EF. The way I understand it GP is an alternative approach to Hessian approximations (GGN, EF) hence having a separate class made more sense to me.

However, this can be easily modified if you prefer to get rid of BackPackGP class. Let me know.

Reply:
Actually, you are right, `BackPackGP` class is superfluous:
- moved kernel methods to `FunctionalLaplace` class. Now they also have a simpler signature since a lot of attributes from `FunctionalLaplace` can be used instead of passing them as params
- moved `gp_quantities` to `BackPackInterface` class

Reply:
Sounds reasonable, I will check out the changes. I don't fully agree regarding GGN/EF not having to do anything with GP: the FullLaplace with GGN is equivalent to the full kernel GP since GGN is what leads to a GP due to linearization. In the same way, one could actually construct a GP corresponding to EF. However, this GP would only be able to predict residuals and is therefore maybe not interesting (except for estimating the marginal likelihood..).